Source/WebCore/Modules/cookie-store/CookieStore.cpp

```diff
@@ -230,6 +231,20 @@ CookieStore::~CookieStore()
     m_mainThreadBridge->detach();
 }
 
+URL CookieStore::creationURL(ScriptExecutionContext& context) const
+{
+    auto url = context.creationURL();
```

Comment:
I think `document.cookieURL()` gives the right value

Reply:
I believe `document.cookieURL()` won't work here. The problem with using `document.url()` is that this returns the document's current URL (`m_url` in Document.h) and this can change with functions like `history.pushState`. And  `m_cookieURL` (which is what `document.cookieURL()` returns) can be set with `m_url`. For example: `setCookieURL(m_url)` in `Document::initSecurityContext()`. So I imagine it's probably best to use the creation URL explicitly as the spec says.

Reply:
Can you try cookieURL and see if it works? This is what WebKit is using for cookies elsewhere so being inconsistent here would be unfortunate (if it can be avoided).

Reply:
Did you test? `Document::initSecurityContext()` should set cookie URL to set parent's url if `shouldInheritSecurityOriginFromOwner`: https://github.com/WebKit/WebKit/blob/280eaf1c4a30cd0650835a410435655ad244f297/Source/WebCore/dom/Document.cpp#L7651

Reply:
Also, I think the new API must use what `document.cookie` is using now, otherwise it's a leak.

Reply:
That's a good point. I hadn't tried it yet (oops)--but I just did, and it works! It looks much cleaner and more consistent this way. Thank you!